src/index.tsx

```diff
@@ -262,18 +263,30 @@ export const WaveCxProvider = (props: {
             <WebView
               source={{ uri: activeContentItem.viewUrl }}
               ref={webViewRef}
-              style={{
-                display: !isRemoteContentReady ? 'none' : undefined,
-              }}
+              style={!isRemoteContentReady ? styles.hidden : undefined}
               onLoad={() => setIsRemoteContentReady(true)}
-              onNavigationStateChange={(event) => {
-                if (
-                  event.url.split('//')[1]?.split('/')[0] !==
-                  activeContentItem?.viewUrl.split('//')[1]?.split('/')[0]
-                ) {
-                  webViewRef.current?.stopLoading();
-                  Linking.openURL(event.url);
+              onShouldStartLoadWithRequest={(request) => {
+                const isInternalLink =
+                  request.url.split('//')[1]?.split('/')[0] ===
+                  activeContentItem?.viewUrl.split('//')[1]?.split('/')[0];
+
+                if (isInternalLink) {
+                  return true;
                 }
+
+                let isDefaultPrevented = false;
+                props.onLinkRequested?.(request.url, {
+                  dismissContent,
+                  preventDefault: () => {
+                    isDefaultPrevented = true;
+                  },
+                });
+
+                if (!isDefaultPrevented) {
+                  Linking.openURL(request.url);
```

Comment:
I guess this is by intention as now any internal url of the dynamic content will be handled outside of the webview allowing WaveCX content to remain within it's originally rendered webview.

Reply:
Yes, intended behavior is:

1. WaveCX content opens directly in the webview. This includes some links beyond the initial content URL -- WaveCX content sometimes links to other WaveCX content and is expected to remain in the web view within the application.
2. Links that are not to WaveCX content are opened in the system browser by default (outside of the integrating application).
3. If an `onLinkRequested` prop is provided and that function invokes the `preventDefault` callback, the WaveCX library _will not do anything_ with the link, as the custom handler is expected to have handled the link as needed.

There is probably room for refactoring this bit of code (specifically I'd like to avoid the mutable `isDefaultPrevented` variable if possible), but it allows a more explicit interface in the custom link handlers where calling `preventDefault` is more clear than e.g. returning a boolean value.

Reply:
Agree, thank you for the explanation!